cmake/Tests.cmake

```diff
@@ -193,16 +193,16 @@ endif()
 if(NOT DEFINED ASAN)
     add_test(
         NAME "cargo-test"
-        COMMAND cargo test ${CARGO_FLAGS} --target-dir target ${cargo_target_opt}
-        WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/fish-rust"
+        COMMAND cargo test ${CARGO_FLAGS} --package fish-rust --target-dir target ${cargo_target_opt}
+        WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
```

Comment:
In an earlier commit I wrongly stated that this change makes CMake workspace aware. As far as I can tell that is wrong, as `cargo` will correctly identify that a crate is a member of a workspace even if run from within the directory of a crate that does not inherit package-attributes/dependencies from the workspace (likely from the lack of a `Cargo.lock`?, not entirely certain), therefore this change essentially only fixes the `target-dir` path to be `fish-shell/target` instead of `fish-shell/fish-rust/target`